Makefile

```diff
@@ -1,323 +1,64 @@
-clean-artifacts:
-	-rm -rf $(INSTALL_PREFIX)
-
-CLEAN_TARGETS += clean-object-files
-.PHONY: clean-object-files
-# Runs go clean to remove any executables or other object files.
-clean-object-files:
-	go clean ./...
-
-CLEAN_TARGETS += clean-vendor
-.PHONY: clean-vendor
-# Removes the ./vendor directory.
-clean-vendor:
-	-rm -rf $(VENDOR_DIR)
-
-CLEAN_TARGETS += clean-tmp
-.PHONY: clean-tmp
-# Removes the ./vendor directory.
-clean-tmp:
-	-rm -rf $(TMP_PATH)
-
-# Keep this "clean" target here after all `clean-*` sub tasks
-.PHONY: clean
-## Cleans the project, removes all generated code/bins and vendor packages
-clean: $(CLEAN_TARGETS)
-
-# -------------------------------------------------------------------
-# build the binary executable (to ship in prod)
-# -------------------------------------------------------------------
+.PHONY: all
+## Builds the operator and runs tests
+all: build test
 
 .PHONY: build
 ## Build the operator
-build: prebuild-check deps check-go-format
-	@echo "building $(BINARY_SERVER_BIN)..."
-	operator-sdk generate k8s
-	go build -v -o $(BINARY_SERVER_BIN) cmd/manager/main.go
-
-# -------------------------------------------------------------------
-# deploy
-# -------------------------------------------------------------------
-
-# to watch all namespaces, keep namespace empty
-APP_NAMESPACE ?= ""
-.PHONY: local
-## Run Operator locally
-local: deploy-rbac build deploy-crd
-	@-oc new-project $(APP_NAMESPACE)
-	operator-sdk up local --namespace=$(APP_NAMESPACE)
-
-.PHONY: deploy-rbac
-## Setup service account and deploy RBAC
-deploy-rbac:
-	@-oc login -u system:admin
-	@-oc create -f deploy/service_account.yaml
-	@-oc create -f deploy/role.yaml
-	@-oc create -f deploy/role_binding.yaml
+build: ./out/operator
 
-.PHONY: deploy-crd
-## Deploy CRD
-deploy-crd:
-	@-oc apply -f deploy/crds/devopsconsole_v1alpha1_component_crd.yaml
-	@-oc apply -f deploy/crds/devopsconsole_v1alpha1_gitsource_crd.yaml
-
-.PHONY: deploy-operator
-## Deploy Operator
-deploy-operator: deploy-crd
-	oc create -f deploy/operator.yaml
-
-.PHONY: deploy-clean
-## Deploy a CR as test
-deploy-clean:
-	@-oc delete component.devopsconsole.openshift.io/myapp
-	@-oc delete imagestream.image.openshift.io/myapp-builder
-	@-oc delete imagestream.image.openshift.io/myapp-output
-	@-oc delete buildconfig.build.openshift.io/myapp-bc
-	@-oc delete deploymentconfig.apps.openshift.io/myapp
-
-.PHONY: deploy-test
-## Deploy a CR as test
-deploy-test:
-	oc create -f examples/devopsconsole_v1alpha1_component_cr.yaml
+.PHONY: clean
+clean:
+	$(Q)-rm -rf ${V_FLAG} ./out
+	$(Q)-rm -rf ${V_FLAG} ./vendor
+	$(Q)go clean ${X_FLAG} ./...
+
+GIT_COMMIT_ID := $(shell git rev-parse HEAD)
+ifneq ($(shell git status --porcelain --untracked-files=no),)
+       GIT_COMMIT_ID := $(GIT_COMMIT_ID)-dirty
+endif
+BUILD_TIME = `date -u '+%Y-%m-%dT%H:%M:%SZ'`
+
+./out/operator: ./vendor $(shell find . -path ./vendor -prune -o -name '*.go' -print)
+	$(Q)operator-sdk generate k8s
```

Comment:
Can we remove this, please? I don't wish to be generating 'critical' code at build time.

Reply:
I did ask you and @corinnekrych  that yesterday and I did get the impression that we have to run this command if the CRD changes. We did the same for fabric8-wit and the design/controller files, no?

Reply:
I had said that actually said the opposite. The operator-sdk always doesn't output reliable content since it's also a development in progress - which is why I prefer a developer to generate it locally, vet it and commit it separately.

Reply:
Okay, sorry for the misunderstanding. I've commented out the generation in 98c35bc.